### PR TITLE
Add new IcoswISC30E3r2 ocean and sea-ice mesh

### PR DESCRIPTION
**Pull request by xylar**
Long name: IcoswISC30L64E3SMv3r2

This Icosahedral (Icos) mesh is made by subdividing an icosahedron and has nearly uniform 30 km resolution.

This mesh is a candidate for the E3SM v3 (E3) low res mesh.  It is revision 2 (r2) of the mesh.  (Revision 1 had incorrect bathymetry and coastline around Greenland.)

The mesh was created using [compass](https://github.com/MPAS-Dev/compass), specifically this tag: https://github.com/MPAS-Dev/compass/releases/tag/mesh_IcoswISC30E3r2

A G-case with data iceberg melt fluxes (DIB) and data ice-shelf melt fluxes (DISMF) has been run for 30 years, with MPAS-Analysis output available here:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis/20230905.G-test.IcoswISC30E3r2.anvil/

The mesh and the G-case results is still been reviewed and approved here:
https://acme-climate.atlassian.net/wiki/spaces/OO/pages/3904831507/Review+IcoswISC30E3r2

A B-case will begin shortly and analysis will be posted here and on the review page as soon as it is available.

### CODE REVIEW

**Comment by xylar** [comment]
This branch is based off of #5927 and will be rebased after that PR is merged.

**Comment by xylar** [comment]
Relevant discussion at https://github.com/E3SM-Ocean-Discussion/E3SM/pull/64

**Comment by xylar** [comment]
@jonbob, what about these files?  Should I sync over the `inputdata` and `diagnostics` files for these, too?

**Review by mark-petersen — APPROVED** [review]
Reviewed mesh, coastlines, critical passages, and MPAS-Analysis simulation results.

**Review by proteanplanet — APPROVED** [review]
Through-flow, shallow bathymetry, cavity geometry and coastlines have been checked.

**Review by erinethomas — APPROVED** [review]
approved - looked at MPAS-analysis and mesh.

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:3299]: Was this made with TempestRemap?

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:3301]: Was this made with TempestRemap?

Comment (xylar) [reply cime_config/config_grids.xml:3301]: @jonbob, that's a question for you.

Comment (jonbob) [reply cime_config/config_grids.xml:3301]: We've been making all conservative "mono" mapping files with tempest, but not the bilinear ones. I'll try to change the process to match the new naming convention and process

**Comment by jonbob** [comment]
20-year B-case using the CRYO1850 compset has completed:
[MPAS-Analysis for: 20230918.test5931.anvil](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.jwolfe/20230918.test5931.anvil/html/)

**Review by darincomeau — APPROVED** [review]

**Review by rljacob — CHANGES_REQUESTED** [review]
Bilinear maps need to be remade with TR.  Map files need to be renamed to follow naming convention.

**Comment by jonbob** [comment]
The new mapping and domain files have been staged to /lcrc/group/e3sm/data/inputdata have been tested successfully with:
* SMS_Ld3.T62_IcoswISC30E3r2.CMPASO-IAF.chrysalis_intel
* SMS_Ld3.TL319_IcoswISC30E3r2.GMPAS-JRA1p5.chrysalis_intel
* SMS_Ld3.ne30pg2_IcoswISC30E3r2.WCYCL1850.chrysalis_intel
* SMS_Ld3.ne30pg2_r05_IcoswISC30E3r2.WCYCL1850.chrysalis_intel
* SMS_Ld3.ne30pg2_r05_IcoswISC30E3r2.WCYCL1850.chrysalis_intel with trfvnp2 mapping file

**Comment by jonbob** [comment]
@rljacob -- can you check that the mapping files are now consistent with the algorithm and naming conventions?

File: cime_config/config_grids.xml
Comment (xylar) [inline cime_config/config_grids.xml:2835]: I'll add a commit to fix the `XXX` here.

**Comment by xylar** [comment]
@jonbob, please verify that the grid files look correct after the rebase.  I don't expect any problems.

@rljacob, could you verify that your concerns have been addressed by @jonbob's latest commit?

**Comment by xylar** [comment]
@alicebarthel, could you either approve or let us know if you have concerns about this mesh?

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:4072]: "patch" should be "esmfpatch"

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:4176]: "patch" should be "esmfpatch"

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:4642]: "smoothed" should be "cstmnn"  for "custom nearest neighbor".

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:4643]: "smoothed" should be "cstmnn"

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:4708]: smoothed should be "cstmnn" for both.

File: cime_config/config_grids.xml
Comment (rljacob) [inline cime_config/config_grids.xml:4798]: "smoothed" should be "cstmnn" for both.

**Comment by rljacob** [comment]
The r2o maps need a name fix but the rest look good.  Also make sure these are on the input data server.

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml
Comment (mark-petersen) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:494]: (documenting a slack conversation) This icos ~30km  should be able to run with exactly the same time step settings as the EC30to60, since it only depends on the minimum cell width. That means
```
<config_dt ocn_grid="IcoswISC30E3r2">'00:30:00'</config_dt>  (it's already set to that)
<config_btr_dt ocn_grid="IcoswISC30E3r2">'0000_00:01:15'</config_btr_dt>
```
The 45s was overly cautious to begin with. 1:15 should work.

File: cime_config/config_grids.xml
Comment (jonbob) [reply cime_config/config_grids.xml]: It should be fixed now

File: cime_config/config_grids.xml
Comment (jonbob) [reply cime_config/config_grids.xml]: should be fixed now

File: cime_config/config_grids.xml
Comment (jonbob) [reply cime_config/config_grids.xml]: should be fixed now -- I also changed the datestamp to YYYYMMDD to match the new naming convention

**Comment by jonbob** [comment]
@rljacob -- I think I fixed all the naming issues, and changed the filenames on the lcrc local inputdata location. Once you approve, I'll copy them to the inputdata server

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml
Comment (mark-petersen) [reply components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml]: OK, thanks. The barotropic time step is limited by an advective CFL condition for the surface gravity waves. There must be some faster velocities on the ICOS30 in locations with 30km cells, where the EC30to60 has larger cells (like mid-latitude western boundary currents). Regardless, we always set this by trial and error in the end. Thanks for testing it. The 1:00 is still a big improvement over 45s.

**Review by rljacob — APPROVED** [review]

**Comment by alicebarthel** [comment]
I approve based on visual inspection of the mesh plots and the G- and B-case MPAS-Analysis plots. Some enhanced biases due to coupling but nothing that look outside of our normal range of issues, or would warrant blocking this configuration in my opinion. Well done!

**Review by alicebarthel — APPROVED** [review]
I approve based on visual inspection of the mesh plots and the G- and B-case MPAS-Analysis plots. Some enhanced biases due to coupling but nothing that look outside of our normal range of issues, or would warrant blocking this configuration in my opinion. Well done!

**Comment by xylar** [comment]
I'm putting this in draft mode because I just discovered a bug in `landIceMask` (which affects the initial condition as well as several mapping files) over the weekend.

**Comment by xylar** [comment]
I'm closing this and will make a new PR for IcoswISC30E3r4 with the bug fix and topography smoothing.

**Pull request closed without merging**
